gridcapa-cse-valid-app/src/main/java/com/farao_community/farao/cse_valid/app/dichotomy/DichotomyRunner.java

```diff
@@ -73,18 +90,24 @@ public DichotomyResult<RaoResponse> runDichotomy(CseValidRequest cseValidRequest
         return engine.run(network);
     }
 
-    private String getJsonCracUrl(CseValidRequest cseValidRequest, Network network) {
+    private String getJsonImportCracUrl(CseValidRequest cseValidRequest, Network network) {
         CseCrac cseCrac = fileImporter.importCseCrac(cseValidRequest.getImportCrac().getUrl());
         Crac crac = fileImporter.importCrac(cseCrac, cseValidRequest.getTimestamp(), network);
         return fileExporter.saveCracInJsonFormat(crac, cseValidRequest.getTimestamp(), cseValidRequest.getProcessType());
     }
 
+    private String getJsonExportCracUrl(CseValidRequest cseValidRequest, Network network) {
```

Comment:
Here too, I'm pretty sure it is possible to mutualise "getJsonImportCracUrl" and "getJsonExportCracUrl" by using a Provider in place of "cseValidRequest.getImportCrac()" ou "cseValidRequest.getExportCrac()".